File: project/frontend/src/types/Empresa.ts

export interface EmpresaCreate {
  id: number;
  nome: string;
  email: string;
  cnpj: string;
  senha: string;
}

export interface Empresa {
  id: number;
  nome: string;
  email: string;
  cnpj: string;
  senha: string;
}

export interface EmpresaUpdate {
  id: number;
  nome: string;
  email: string;
  cnpj: string;
  senha: string;
}

File: project/backend/src/main/java/com/sistema/moedaEstudantil/models/Empresa.java
package com.sistema.moedaEstudantil.models;

import com.sistema.moedaEstudantil.dto.EmpresaDTO;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Inheritance(strategy = InheritanceType.JOINED)
public class Empresa extends Usuario {

    @Column(name = "cnpj", length = 18, unique = true)
    private String cnpj;

    public EmpresaDTO toDTO() {
        return new EmpresaDTO(
                id, nome, cnpj, email, senha
        );
    }
}